modules/bash-commons/src/assert.sh

```diff
@@ -70,6 +70,35 @@ function assert_value_in_list {
   fi
 }
 
+function assert_exactly_one_of {
+  local -r arg_name1="$1"
+  local -r arg_val1="$2"
+  local -r arg_name2="$3"
+  local -r arg_val2="$4"
+  local -r arg_name3="$5"
+  local -r arg_val3="$6"
```

Comment:
Could just read in an array of args and loop over them in pairs to make this more flexible? Otherwise, it only works for up to 3 arguments, which seems a bit limited.

Reply:
Rewrote to support array input.

Reply:
... and added a few more tests.